### PR TITLE
New option that prevents publishing blocked tests

### PR DESCRIPTION
**Pull request by apallier**
This PR implements a new option `--tr-dont-publish-blocked` that prevents publishing blocked tests in TestRail.

Use case:
I have an existing testrun in TestRail with blocked tests. I know these blocked tests should not be executed anymore (known bugs, feature not ready...) and I would like to prevent pytest to publish "failed" status on these tests.

### CODE REVIEW

File: pytest_testrail/plugin.py [hunk 1]
```diff
@@ -222,6 +224,17 @@ def add_results(self, testrun_id):
         self.results.sort(key=itemgetter('status_id'))
         self.results.sort(key=itemgetter('case_id'))
 
+        # Manage case of "blocked" testcases
+        if self.publish_blocked is False:
+            print('[{}] Option "Don\'t publish blocked testcases" activated'.format(TESTRAIL_PREFIX))
+            blocked_tests_list = [
+                test.get('case_id') for test in self.get_tests(testrun_id)
+                if test.get('status_id') == 2
```
Comment (allankp) [inline pytest_testrail/plugin.py]: Blocked and Skipped are the same status ID in testrail ?
Can this be defined in the status dict at the top.

File: pytest_testrail/plugin.py [hunk 2]
```diff
@@ -365,3 +378,18 @@ def get_available_testruns(self, plan_id):
...
+    def get_tests(self, run_id):
+        """
+        :return: the list of tests containing in a testrun.
+
+        """
+        response = self.client.send_get(
+            GET_TESTS_URL.format(run_id),
+            cert_check=self.cert_check
+        )
+        error = self.client.get_error(response)
```
Comment (allankp) [inline pytest_testrail/plugin.py]:
```
    if  self.client.get_error(response):
        print('[{}] Failed to get tests: "{}"'.format(TESTRAIL_PREFIX, error))
        return
return response
```
Would that work and be a bit cleaner ?

File: pytest_testrail/plugin.py (hunk 1 above)
Comment (apallier) [reply pytest_testrail/plugin.py]: 2 is the status of "Blocked" in TestRail. "Skipped" status doesn't exist in TestRail. I don't know why mapping between "skipped" in Pytest and "Blocked" has been made in `PYTEST_TO_TESTRAIL_STATUS` but I don't use it in this case.
I am dealing only with TestRail status here.

File: pytest_testrail/plugin.py (hunk 2 above)
Comment (apallier) [reply pytest_testrail/plugin.py]: Ok, I use implicit behavior of Python functions that return always `None` if no explicit `return` occurred

**Comment by allankp** [comment]
@apallier can you merge master in please

**Pull request merged**